src/plugins/content_management/server/event_stream/memory/memory_event_stream_client.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  EventStreamClient,
+  EventStreamClientFilterOptions,
+  EventStreamClientFilterResult,
+  EventStreamEvent,
+} from '../types';
+import { clone } from './util';
+
+/**
+ * This is an in-memory implementation of the {@link EventStreamClient}
+ * interface (it does not persist events to Elasticsearch). It is useful for
+ * testing and demo purposes.
+ */
+export class MemoryEventStreamClient implements EventStreamClient {
+  #events: EventStreamEvent[] = [];
+
+  public async initialize(): Promise<void> {}
+
+  public async writeEvents(events: EventStreamEvent[]): Promise<void> {
+    for (const event of events) {
+      this.#events.push(clone(event));
+    }
+    this.#events.sort((a, b) => b.time - a.time);
```

Comment:
I suspect this will get expensive, over time :-) - but it's not used in production IIUC, so ... "it's fine".  But this could be optimized (assuming everything is already sorted) if required ...

Reply:
You are right, inserting it into the right position on *write* is probably faster (assuming few writes and more reads) then sorting the whole thing on *read*. But this class is used only for testing and event with tens of thousands of items it would be good enough for testing.